packages/waku/src/lib/builder/build.ts

```diff
@@ -559,104 +562,112 @@ const emitHtmlFiles = async (
-        let htmlHead = publicIndexHtmlHead;
-        if (cssAssets.length) {
-          const cssStr = cssAssets
-            .map(
-              (asset) =>
-                `<link rel="stylesheet" href="${config.basePath}${asset}">`,
-            )
-            .join('\n');
-          // HACK is this too naive to inject style code?
-          htmlStr = htmlStr.replace(/<\/head>/, cssStr);
-          htmlHead += cssStr;
-        }
-        const rscPathsForPrefetch = new Set<string>();
-        const moduleIdsForPrefetch = new Set<string>();
-        for (const { rscPath, skipPrefetch } of entries || []) {
-          if (!skipPrefetch) {
-            rscPathsForPrefetch.add(rscPath);
-            for (const id of getClientModules(rscPath)) {
-              moduleIdsForPrefetch.add(id);
-            }
-          }
-        }
-        const code =
-          generatePrefetchCode(
-            basePrefix,
-            rscPathsForPrefetch,
-            moduleIdsForPrefetch,
-          ) + (customCode || '');
-        if (code) {
-          // HACK is this too naive to inject script code?
-          htmlStr = htmlStr.replace(
-            /<\/head>/,
-            `<script type="module" async>${code}</script></head>`,
-          );
-          htmlHead += `<script type="module" async>${code}</script>`;
-        }
-        if (!isStatic) {
-          dynamicHtmlPathMap.set(pathSpec, htmlHead);
-          return;
-        }
-        pathname = pathSpec2pathname(pathSpec);
-        const destHtmlFile = joinPath(
-          rootDir,
-          config.distDir,
-          DIST_PUBLIC,
-          extname(pathname)
-            ? pathname
-            : pathname === '/404'
-              ? '404.html' // HACK special treatment for 404, better way?
-              : pathname + '/index.html',
-        );
-        // In partial mode, skip if the file already exists.
-        if (existsSync(destHtmlFile)) {
-          return;
-        }
-        const htmlReadable = await renderHtml({
-          config,
-          pathname,
-          searchParams: new URLSearchParams(),
-          htmlHead,
-          renderRscForHtml: (rscPath, rscParams) =>
-            renderRsc(
-              { env, config, rscPath, context, decodedBody: rscParams },
-              { isDev: false, entries: distEntries },
-            ),
-          getSsrConfigForHtml: (pathname, searchParams) =>
-            getSsrConfig(
-              { env, config, pathname, searchParams },
-              { isDev: false, entries: distEntries },
-            ),
-          isDev: false,
-          loadModule: distEntries.loadModule,
-        });
-        await mkdir(joinPath(destHtmlFile, '..'), { recursive: true });
-        if (htmlReadable) {
-          await pipeline(
-            Readable.fromWeb(htmlReadable as any),
-            createWriteStream(destHtmlFile),
-          );
-        } else {
-          await writeFile(destHtmlFile, htmlStr);
+  const handlePage = async ({
```

Comment:
We don't want to use the term "Page" here. Can it be something like `handlePath`?

Reply:
changed to handlePath ✅